### PR TITLE
chore: use 11.0.3 CLI for testing

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

Use the CLI `11.0.3` for testing our changes locally and in CI.
This avoids the problem where CLI prompts for 2 available versions of API plugin - `ownerpkfix...` tag i.e (consumed in CLI `11.0.4+`) and `5.2.0` i.e current.

#### Description of how you validated changes
Full e2e running, status is reported in the github checks.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "update-versions": "lerna version --yes --no-commit-hooks --no-push --exact --conventional-commits --no-git-tag-version",
     "commit": "git-cz",
     "coverage": "codecov || exit 0",
-    "add-cli-no-save": "yarn add @aws-amplify/cli-internal -W  && git checkout -- package.json",
+    "add-cli-no-save": "yarn add @aws-amplify/cli-internal@11.0.3 -W  && git checkout -- package.json",
```
Comment (sundersc) [inline package.json:42]: Should we change the CLI version in CCI config files?

Comment (phani-srikar) [reply package.json:42]: We need another change there. On it

Comment (sundersc) [reply package.json:42]: We should update the config.yml as well.

**Review by AaronZyLee — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -140,7 +140,8 @@
     "parse-url": "^8.1.0",
     "moment-timezone": "^0.5.35",
     "**/aws-cdk-lib": "~2.68.0",
-    "**/@aws-amplify/amplify-cli-core": "^4.0.0",
+    "**/@aws-amplify/amplify-cli-core": "4.0.1",
+    "**/amplify-cli-core": "4.0.1",
```
Comment (dpilch) [inline package.json:144]: Why is the non-scoped version needed in resolutions?

**Review by dpilch — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

File: packages/amplify-category-api/package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
     "uuid": "^8.3.2"
   },
   "peerDependencies": {
-    "@aws-amplify/amplify-cli-core": "^4.0.1",
+    "@aws-amplify/amplify-cli-core": "4.0.1",
```
Comment (AaronZyLee) [inline packages/amplify-category-api/package.json:76]: TODO: we might need revert to the caret dependency once the issue is resolved

**Review by AaronZyLee — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (phani-srikar) [reply package.json:144]: Probably not needed, just to be extra sure.

**Pull request merged**